src/sdk/list.hpp

```diff
@@ -0,0 +1,404 @@
+#ifndef realm_sdk_list_hpp
+#define realm_sdk_list_hpp
+
+#include "accessor.hpp"
+#include "property.hpp"
+#include "type_info.hpp"
+#include <type_traits>
+
+namespace realm::sdk {
+
+template <typename T>
+struct Object;
+
+// MARK: List
+
+template <class T>
+struct Property<std::vector<T>, typename std::enable_if_t<!std::is_pointer_v<T>>>
+: private PropertyBase
+{
+    static_assert(!std::is_base_of_v<Object<T>, T>, "Cannot create List for Object Type");
+
+    using base = PropertyBase;
+    using base::base;
+    using base::is_managed;
+
+    void push_back(T&& value)
+    {
+        if (!base::is_managed()) {
+            return m_value.push_back(std::move(value));
+        }
+
+        base::m_accessor.as<std::vector<T>>().add(value);
+    }
+
+    void push_back(T& value)
+    {
+        if (!base::is_managed()) {
+            return m_value.push_back(value);
+        }
+
+        base::m_accessor.as<std::vector<T>>().add(value);
+    }
+
+    T& at(size_t idx)
+    {
+        if (!base::is_managed()) {
+            return m_value.at(idx);
+        }
+
+        m_value[idx] = base::m_accessor.as<std::vector<T>>().get(idx);
```

Comment:
Everything involving `m_value` for managed lists seems invalid. There's nothing here that guarantees that `m_value.size() > idx` here. Returning a reference to an entry in `m_value` just allows nonsense (e.g. `list.at(0) += 1` will compile and mutate a value that can never be read again). The reference returned is only valid until the next index is read, so there's also no need to store anything more than a single value.